libs/extended/src/System/Logger/Extended.hs

```diff
@@ -77,6 +78,43 @@ collect = foldr go (Element' mempty [])
 jsonRenderer :: Renderer
 jsonRenderer _sep _dateFormat _logLevel = fromEncoding . elementToEncoding . collect
 
+data StructuredJSONOutput = StructuredJSONOutput {msgs :: [Text], fields :: Map Text [Text]}
+
+-- | Displays all the 'Bytes' segments in a list under key @msgs@ and 'Field'
+-- segments as key-value pair in a JSON
+--
+-- >>> logElems = [Bytes "I", Bytes "The message", Field "field1" "val1", Field "field2" "val2", Field "field1" "val1.1"]
+-- >>> B.toLazyByteString $ structuredJSONRenderer "," iso8601UTC Info logElems
+-- "{\"msgs\":[\"I\",\"The message\"],\"field1\":[\"val1\",\"val1.1\"],\"field2\":\"val2\",\"level\":\"Info\"}"
+structuredJSONRenderer :: Renderer
+structuredJSONRenderer _sep _dateFmt lvl logElems =
+  let structuredJSON = toStructuredJSONOutput logElems
+   in fromEncoding . toEncoding $
+        object
+          ( [ "level" Aeson..= lvl,
```

Comment:
@jschaul This line adds the level to the final JSON.

Reply:
Cool, thanks!